src/DynamoCore/Search/NodeSearchModel.cs

```diff
@@ -48,58 +50,110 @@ internal XmlDocument ComposeXmlForLibrary()
                 entry => entry.Categories);
 
             foreach (var category in root.SubCategories)
-                AddCategoryToXml(document.DocumentElement, category);
+                AddCategoryToXml(document.DocumentElement, category, dynamoPath);
 
             foreach (var entry in root.Entries)
-                AddEntryToXml(document.DocumentElement, entry);
+                AddEntryToXml(document.DocumentElement, entry, dynamoPath);
 
             return document;
         }
 
-        private static void AddEntryToXml(XmlNode parent, NodeSearchElement entry)
+        private static void AddEntryToXml(XmlNode parent, NodeSearchElement entry, string dynamoPath)
         {
             var element = XmlHelper.AddNode(parent, entry.GetType().ToString());
             XmlHelper.AddNode(element, "FullCategoryName", entry.FullCategoryName);
             XmlHelper.AddNode(element, "Name", entry.Name);
             XmlHelper.AddNode(element, "Group", entry.Group.ToString());
             XmlHelper.AddNode(element, "Description", entry.Description);
 
+            // Add search tags, joined by ",".
+            // E.g. <SearchTags>bounding,bound,bymaxmin,max,min,bypoints</SearchTags>
+            XmlHelper.AddNode(element, "SearchTags",
+                String.Join(",", entry.SearchKeywords.Where(word => !String.IsNullOrWhiteSpace(word))));
+
+            var dynamoNode = entry.CreateNode();
+
             // If entry has input parameters.
-            if (entry.InputParameters.First().Item2 != Properties.Resources.NoneString)
+            if (dynamoNode.InPorts.Count != 0)
             {
                 var inputNode = XmlHelper.AddNode(element, "Inputs");
-                foreach (var parameter in entry.InputParameters)
+                for (int i = 0; i < dynamoNode.InPorts.Count; i++)
                 {
                     var parameterNode = XmlHelper.AddNode(inputNode, "InputParameter");
-                    XmlHelper.AddAttribute(parameterNode, "Name", parameter.Item1);
-                    XmlHelper.AddAttribute(parameterNode, "Type", parameter.Item2);
+                    XmlHelper.AddAttribute(parameterNode, "Name", dynamoNode.InPorts[i].PortName);
+
+                    // Case for UI nodes as ColorRange.
+                    if (dynamoNode.InPorts.Count == entry.InputParameters.Count()
+                        && entry.InputParameters.First().Item2 != Properties.Resources.NoneString)
```

Comment:
No, here everything is written correct. We have little problem with UI nodes.
Some of them do not have incoming ports in `NodeSearchElement`, but do have incoming ports in `NodeModel`.
E.g. node List.Create.
List initializes its incoming ports, when constructor is called. So, here we check if nodeModel has input ports and nodeSearchElement also has the same input ports.
In case of list.create node it will have several incoming ports on nodeModel, but 0 incoming ports on nodeSearchElement.